views_api.py

```diff
@@ -230,6 +239,14 @@ async def api_reverse_submarineswap_create(
         raise HTTPException(
             status_code=HTTPStatus.METHOD_NOT_ALLOWED, detail="Insufficient balance."
         )
+    settings = await get_or_create_boltz_settings()
+    try:
+        validate_address(data.onchain_address, settings.boltz_network)
+    except ValueError as exc:
+        raise HTTPException(
+            status_code=HTTPStatus.METHOD_NOT_ALLOWED,
```

Comment:
Why are you throwing not allowed when parameters are wrong?

> [405 Method Not Allowed](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405)
> The request method is known by the server but is not supported by the target resource. For example, an API may not allow calling DELETE to remove a resource.

Reply:
any other suggestion is use that mainly because it shows a warning instead of error on the frontend.